src/compiler/transformers/declarations.ts

```diff
@@ -695,7 +696,7 @@ export function transformDeclarations(context: TransformationContext) {
             if (elem.kind === SyntaxKind.OmittedExpression) {
                 return elem;
             }
-            if (elem.propertyName && isIdentifier(elem.propertyName) && isIdentifier(elem.name) && !elem.symbol.isReferenced) {
+            if (elem.propertyName && isIdentifier(elem.propertyName) && isIdentifier(elem.name) && !elem.symbol.isReferenced && !identifierToKeywordKind(elem.propertyName)) {

```

Comment:
I feel like this needs to be `isIdentifierANonContextualKeyword` or something; can you try out a contextual keyword like `abstract` and see what happens?

Reply:
`await` (and maybe `async`) probably also another good one to check.

Reply:
Yeah, none of `async`, `await`, or `abstract` need to be renamed in this context, so definitely need to check if it's a contextual keyword or not.

Reply:
I think `isIdentifierANonContextualKeyword` doesn't handle `enum`, `function`, etc.

Reply:
Ok